company-ycmd.el

```diff
@@ -104,26 +104,20 @@ feature."
 
 (defun company-ycmd--prefix-candidate-p (candidate prefix)
   "Return t if CANDIDATE string begins with PREFIX."
-  (let ((insertion-text (assoc-default 'insertion_text candidate))
-        (case-fold-search t))
-    (equal (string-match (regexp-quote prefix) insertion-text) 0)))
-
-(defmacro company-ycmd--with-destructured-candidate (candidate prefix start-col body)
-  "Destructure CANDIDATE and evalueate BODY.
-
-PREFIX and START-COL are used to calculate the insertion text."
-  (declare (indent 3) (debug t))
-  `(let* ((insertion-text (assoc-default 'insertion_text candidate))
-          (detailed-info (assoc-default 'detailed_info candidate))
-          (kind (assoc-default 'kind candidate))
-          (extra-menu-info (assoc-default 'extra_menu_info candidate))
-          (menu-text (assoc-default 'menu_text candidate))
-          (extra-data (assoc-default 'extra_data candidate))
-          (prefix-start-col (- (+ 1 (ycmd--column-in-bytes)) (length prefix)))
-          (prefix-size (- start-col prefix-start-col))
-          (item (concat (substring-no-properties prefix 0 prefix-size)
-                        (substring-no-properties
-                         (s-chop-suffix "()" insertion-text)))))
+  (let ((insertion-text (assoc-default 'insertion_text candidate)))
+    (s-starts-with? prefix insertion-text t)))
+
+(defmacro company-ycmd--with-destructured-candidate (candidate body)
```

Comment:
interface is also more simple and concise now by extracting all information from candidate.